src/Bicep.Core/Intermediate/ExpressionBuilder.cs

```diff
@@ -132,7 +132,7 @@ private Expression ConvertWithoutLowering(SyntaxBase syntax)
             case ResourceAccessSyntax resourceAccess:
                 return ConvertResourceAccess(resourceAccess);
             case LambdaSyntax lambda:
-                var variables = lambda.GetLocalVariables();
+                var variables = lambda.GetLocalVariables().ToArray();
```

Comment:
Should we just have `GetLocalVariables` return a collection rather than enumerable?

Reply:
Makes sense. Changed it to return `ImmutableArray`.